### DISCUSSION TITLE
How to host as Azure App Service with custom domain?

### DISCUSSION THREAD

**Discussion by luiscantero** [datenstrom/community#678] [Ask a question] [answered]
Hi,
I've run into an issue using v0.8.52 while trying to host yellow as a Linux App Service on Microsoft Azure:
* When hosting a web site on Azure, by default it gets a name such as "example.azurewebsites.net". Then you can map a custom domain on top of it, e.g. www.example.com.
* I did the above and edited the yellow-system.ini to reflect the www.example.com URL. However, the meta og:url and also the edit link where still pointing to example.azurewebsites.net. I was also unable to login to the edit area.
* This is because `core.php` uses the environment variable `SERVER_NAME`, which on Azure still points to example.azurewebsites.net.
* I was able to fix the issue by replacing `SERVER_NAME` with `HTTP_X_FORWARDED_HOST`. However, I think it would be better to honor the user setting in yellow-system.ini.

Or is there a better way to deal with this issue?

Thanks,
Luis

**Comment by luiscantero** [comment]
Ping: As written above, I already found a solution to this problem. My request is for a code change so that yellow will use the server name from the yellow-system.ini instead of the one from the environment variable SERVER_NAME.
Does it make sense?
Thanks.

**Comment by markseuffert** [comment, marked as answer]
I am afraid to say that I am not familiar with Azure App Service. Datenstrom Yellow works both as content management system (CMS) and static site generator (SSG). Usually the URL of the website is automatically detected, but if this doesn't work you can always pin down the URL in your system settings:

`CoreServerUrl` = URL of the website, when used as a content management system (CMS).
`CommandStaticUrl` = URL of the website, when used as a static site generator (SSG).

See also https://github.com/annaesvensson/yellow-core#settings.

Hope this helps.

**Reply by luiscantero** [reply]
Thanks a lot, that solved my problem.

**Comment by markseuffert** [comment]
Last week someone mentioned problems with a web server behind a reverse proxy… and that might be related to this discussion.

I think the underlying problem is that the server URL is not automatically detected, at least not in all cases. For example when the web server is behind a reverse proxy, load balancer or TLS termination. Maybe the source code could be improved for handling more scenarios, but at the moment we do not have access to a good test environment. The technical documentation and RFCs [[1](https://www.rfc-editor.org/rfc/rfc3875)][[2](https://www.rfc-editor.org/rfc/rfc7239)] are a bit thin when it comes to CGI variables like `HTTP_X_FORWARDED-*` and `FORWARDED`. Unfortunately we can not blindly trust these HTTP request headers. They may come from malicious HTTP clients. So it might not even be possible to cover all scenarios. Even if we wanted to. In the worst case the installer might have to detect reverse proxies and then tell the user what to do.

If someone runs into this problem in the future here's what you have to look for. During installation the [installer](https://github.com/annaesvensson/yellow-install) shows the error message `Datenstrom Yellow requires rewrite support!`. This could mean one of two things right now a) web server configuration is not working... or... b) web server is behind a reverse proxy and the server URL is not automatically detected. In the later case the solution is the same as mentioned in this discussion, to manually configure `CoreServerUrl` in your [system settings](https://github.com/annaesvensson/yellow-core#settings).

Anyone has experience with reverse proxies? Any thoughts?
